bazel/linux/defs.bzl

```diff
@@ -655,3 +655,181 @@ The test will run locally inside a user-mode linux process.
     },
     test = True,
 )
+
+NvDriverInfo = provider(
+    doc = """Maintains the information necessary to represent the nvidia driver kernel modules.""",
+    fields = {
+        "name": "Name of the rule that defined this driver.",
+        "package": "A string indicating which package this driver has been built against. For example, 'centos-kernel-5.3.0-1'.",
+        "modules": "An array of files representing the compiled kernel modules (.ko).",
+    },
+)
+
+def _nv_driver(ctx):
```

Comment:
both `nv_driver` and `nv_driver_rule` look very similar to `kernel_module` and `kernel_module_rule`. Do we need a dedicated nvidia rule? or can we just generalize the `kernel_module_rule` to eg, take either a list or a single module name? take extra parameters for the makefile generation?

Reply:
I can give it a go

Reply:
I added a new commit that generalizes the kernel module rules.  These existing targets all build fine:

`bazel build //sdk/driver:enf`
`bazel build //sdk/driver:enf_queues_test`
`bazel build //poc/ibuverbs/driver:all`

Along with a new `//poc/nv-fpga/driver:all` target that is WIP.

However, the current patch breaks the kunit test.  I get this now:
```
(default-dev-env)curt@enfabrica:~/src/internal$ bazel build //sdk/driver:enf_queues_unit_test
INFO: Invocation ID: 311c40b1-f2fc-4868-bd17-76e82eea9122
ERROR: /home/curt/src/internal/sdk/driver/BUILD.bazel:42:12: in module attribute of kernel_test rule //sdk/driver:enf_queues_unit_test: '//sdk/driver:enf_queues_test' does not have mandatory providers: 'KernelModulesInfo'
ERROR: Analysis of target '//sdk/driver:enf_queues_unit_test' failed; build aborted: Analysis of target '//sdk/driver:enf_queues_unit_test' failed
INFO: Elapsed time: 2.903s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)
```

Reply:
I understand why the kunit test broke.  I will add commit soon to address it.